premarum-backend/PreEnrollmentMgmt.Core/Entities/PreEnrollment.cs

```diff
@@ -45,4 +46,27 @@ public void RemoveSelections(int[] CourseOfferings)
     {
         ((HashSet<SemesterOffer>) Selections).RemoveWhere( so => CourseOfferings.Contains(so.Id));
     }
+
+    public bool HasMoreThan21Credits()
+    {
+        int credits = Selections
+            .Select(so => so.Course.CourseCredit)
+            .Sum();
+        return credits > 21;
```

Comment:
Later we could take this value from environment variables as something like CONFIGURATION_CREDIT_LIMIT or any name we deem appropriate.